pkg/generate/app/imagelookup.go

```diff
@@ -133,64 +121,79 @@ func (r DockerClientResolver) lookup(value string) (*ComponentMatch, error) {
+	componentMatches := ComponentMatches{}
+	for _, term := range terms {
+		ref, err := imageapi.ParseDockerImageReference(term)
+		if err != nil {
+			return nil, err
 		}
-		return nil, ErrNoMatch{value: value, qualifier: fmt.Sprintf("can't connect to %q: %v", ref.Registry, err)}
-	}
-	image, err := connection.ImageByTag(ref.Namespace, ref.Name, ref.Tag)
-	if err != nil {
-		if dockerregistry.IsNotFound(err) {
-			return nil, ErrNoMatch{value: value, qualifier: err.Error()}
+
+		glog.V(4).Infof("checking Docker registry for %q", ref.String())
+		connection, err := r.Client.Connect(ref.Registry, r.AllowInsecure)
+		if err != nil {
+			if dockerregistry.IsRegistryNotFound(err) {
+				return nil, ErrNoMatch{value: term}
+			}
+			return nil, fmt.Errorf("can't connect to %q: %v", ref.Registry, err)
+		}
+
+		image, err := connection.ImageByTag(ref.Namespace, ref.Name, ref.Tag)
+		if err != nil {
+			if dockerregistry.IsNotFound(err) {
+				return nil, ErrNoMatch{value: term, qualifier: err.Error()}
+			}
+			return nil, fmt.Errorf("can't connect to %q: %v", ref.Registry, err)
+		}
+
+		if len(ref.Tag) == 0 {
+			ref.Tag = imageapi.DefaultImageTag
+		}
+		glog.V(4).Infof("found image: %#v", image)
+
+		dockerImage := &imageapi.DockerImage{}
+		if err = kapi.Scheme.Convert(image, dockerImage); err != nil {
+			return nil, err
+		}
+
+		if len(ref.Registry) == 0 {
+			ref.Registry = "Docker Hub"
 		}
-		return nil, ErrNoMatch{value: value, qualifier: fmt.Sprintf("can't connect to %q: %v", ref.Registry, err)}
-	}
-	if len(ref.Tag) == 0 {
-		ref.Tag = imageapi.DefaultImageTag
-	}
-	glog.V(4).Infof("found image: %#v", image)
-	dockerImage := &imageapi.DockerImage{}
-	if err = kapi.Scheme.Convert(image, dockerImage); err != nil {
-		return nil, err
-	}
 
-	if len(ref.Registry) == 0 {
-		ref.Registry = "Docker Hub"
+		componentMatches = append(componentMatches, &ComponentMatch{
+			Value:       term,
+			Argument:    fmt.Sprintf("--docker-image=%q", term),
+			Name:        term,
+			Description: descriptionFor(dockerImage, term, ref.Registry, ref.Tag),
+			Builder:     IsBuilderImage(dockerImage),
+			Score:       0,
+			Image:       dockerImage,
+			ImageTag:    ref.Tag,
+		})
 	}
-	return &ComponentMatch{
-		Value:       value,
-		Argument:    fmt.Sprintf("--docker-image=%q", value),
-		Name:        value,
-		Description: descriptionFor(dockerImage, value, ref.Registry),
-		Builder:     IsBuilderImage(dockerImage),
-		Score:       0,
-		Image:       dockerImage,
-		ImageTag:    ref.Tag,
-	}, nil
+
+	return componentMatches, nil
 }
 
-func descriptionFor(image *imageapi.DockerImage, value, from string) string {
+func descriptionFor(image *imageapi.DockerImage, value, from string, tag string) string {
 	shortID := image.ID
 	if len(shortID) > 7 {
 		shortID = shortID[:7]
 	}
-	parts := []string{fmt.Sprintf("Docker image %q", value), shortID, fmt.Sprintf("from %s", from)}
+	tagPart := ""
+	if len(tag) > 0 {
```

Comment:
I guess if the image doesn't have one specified it's ok not to output its description with 'latest'